server/src/main/java/org/apache/iotdb/db/metadata/storagegroup/StorageGroupManager.java

```diff
@@ -0,0 +1,259 @@
+    mtree.deleteStorageGroup(storageGroup);
+  }
+
+  @Override
+  public boolean isStorageGroup(PartialPath path) {
+    return mtree.isStorageGroup(path);
+  }
+
+  @Override
+  public boolean checkStorageGroupByPath(PartialPath path) {
+    return mtree.checkStorageGroupByPath(path);
+  }
+
+  @Override
+  public PartialPath getBelongedStorageGroup(PartialPath path) throws StorageGroupNotSetException {
+    return mtree.getBelongedStorageGroup(path);
+  }
+
+  @Override
+  public List<PartialPath> getBelongedStorageGroups(PartialPath pathPattern)
+      throws MetadataException {
+    return mtree.getBelongedStorageGroups(pathPattern);
+  }
+
+  @Override
+  public List<PartialPath> getMatchedStorageGroups(PartialPath pathPattern, boolean isPrefixMatch)
+      throws MetadataException {
+    return mtree.getMatchedStorageGroups(pathPattern, isPrefixMatch);
+  }
+
+  @Override
+  public List<PartialPath> getAllStorageGroupPaths() {
+    return mtree.getAllStorageGroupPaths();
+  }
+
+  @Override
+  public Map<String, List<PartialPath>> groupPathByStorageGroup(PartialPath path)
+      throws MetadataException {
+    return mtree.groupPathByStorageGroup(path);
+  }
+
+  @Override
+  public int getStorageGroupNum(PartialPath pathPattern, boolean isPrefixMatch)
+      throws MetadataException {
+    return mtree.getStorageGroupNum(pathPattern, isPrefixMatch);
+  }
+
+  @Override
+  public IStorageGroupMNode getStorageGroupNodeByStorageGroupPath(PartialPath path)
+      throws MetadataException {
+    return mtree.getStorageGroupNodeByStorageGroupPath(path);
+  }
+
+  @Override
+  public IStorageGroupMNode getStorageGroupNodeByPath(PartialPath path) throws MetadataException {
+    return mtree.getStorageGroupNodeByPath(path);
+  }
+
+  @Override
+  public List<IStorageGroupMNode> getAllStorageGroupNodes() {
+    return mtree.getAllStorageGroupNodes();
+  }
+
+  @Override
+  public boolean isPathExist(PartialPath path) {
+    return mtree.isPathExist(path);
+  }
+
+  @Override
+  public Pair<Integer, List<SGMManager>> getNodesCountInGivenLevel(
+      PartialPath pathPattern, int level, boolean isPrefixMatch) throws MetadataException {
+    Pair<Integer, Set<IStorageGroupMNode>> resultAboveSG =
+        mtree.getNodesCountInGivenLevel(pathPattern, level, isPrefixMatch);
+    List<SGMManager> sgmManagers = new LinkedList<>();
+    for (IStorageGroupMNode storageGroupMNode : resultAboveSG.right) {
+      sgmManagers.add(storageGroupMNode.getSGMManager());
+    }
+    return new Pair<>(resultAboveSG.left, sgmManagers);
+  }
+
+  @Override
+  public Pair<List<PartialPath>, List<SGMManager>> getNodesListInGivenLevel(
+      PartialPath pathPattern, int nodeLevel, MManager.StorageGroupFilter filter)
+      throws MetadataException {
+    Pair<List<PartialPath>, Set<IStorageGroupMNode>> resultAboveSG =
+        mtree.getNodesListInGivenLevel(pathPattern, nodeLevel, filter);
+    List<SGMManager> sgmManagers = new LinkedList<>();
```

Comment:
Why does the `resultAboveSG.left` is needed ？
From my understanding, the MTreeAboveSG does not contain the detailed series.

Reply:
This feature manages the mnode in MTree, including measurementMNodes, thus there may be matched results in MTreeAboveSG.

Reply:
This method is used by the old cluster to finish task of  ```getMeasurementGroupByLevel```. This may be removed in future development of new cluster.